app/controllers/file.rb

```diff
@@ -57,6 +57,10 @@ def slim(template, options={})
 
 				get "/?*", if: -> { resource.directory? || !resource.exist? } do
 					authorize resource, :index?
+
+					session["moves"] ||= {}
+					session["moves"][site.route] ||= []
```

Comment:
could be, but what if `session["moves"]` already exists from having been set on a different site?  Then you would still need to add the `[site.route]` key